tasks/go_deps.py

```diff
@@ -185,3 +186,95 @@ def show(ctx: Context, build: str, flavor: str = AgentFlavor.base.name, os: str
 
     for dep in deps:
         print(dep)
+
+
+@task(
+    help={
+        'build': f'The agent build to use, one of {", ".join(BINARIES.keys())}',
+        'flavor': f'The agent flavor to use, one of {", ".join(AgentFlavor.__members__.keys())}. Defaults to base',
+        'os': f'The OS to use, one of {", ".join(GOOS_MAPPING.keys())}. Defaults to host platform',
+        'arch': f'The architecture to use, one of {", ".join(GOARCH_MAPPING.keys())}. Defaults to host architecture',
+        'entrypoint': 'A Go package path, defaults to the entrypoint of the build.',
+        'target': 'A Go package path. If specified, generate a graph from the entrypoint to the target.',
+        'std': 'Whether to include the standard library in the graph',
+        'cluster': 'Whether to group packages by cluster.',
+        'fmt': 'The format of the generated image. Must be accepted by "dot -T". Defaults to "svg".',
+        'auto_open': 'Whether to open the generated graph automatically.',
+    }
+)
+def graph(
+    ctx: Context,
+    build: str,
+    flavor: str = AgentFlavor.base.name,
+    os: str | None = None,
+    arch: str | None = None,
+    entrypoint: str | None = None,
+    target: str | None = None,
+    std: bool = False,
+    cluster: bool = False,
+    fmt: str = "svg",
+    auto_open: bool = True,
+):
+    """
+    Generate a dependency graph of the given build.
+    Requires https://github.com/loov/goda and Graphviz's `dot` tools to be in the PATH.
+
+    Usage:
+        Dependency graph of the trace-agent on Linux
+          inv -e go-deps.graph --build trace-agent --os linux
+        Reachability graph of the process-agent on Linux to k8s.io/... dependencies
+          inv -e go-deps.graph --build process-agent --os linux --target k8s.io/...
+        Dependency graph of workloadmeta on Linux when using the same build tags as the core-agent
+          inv -e go-deps.graph --build agent --os linux --entrypoint github.com/DataDog/datadog-agent/comp/core/workloadmeta/...:all
+    """
+    if os is None:
+        goos = ctx.run("go env GOOS", hide=True)
+        assert goos
+        os = goos.stdout.strip()
+    else:
+        assert os in GOOS_MAPPING.values()
```

Comment:
```suggestion
    os = os or ctx.run("go env GOOS", hide=True).stdout.strip()
```
same for `arch` value, old memory from doing some python, may need to be tested

Reply:
You can add the assertion just below, I forget them

Reply:
I agree but to please the type system I need to assert the return value of `ctx.run` is not `None`